src/fillTank.test.js

```diff
@@ -1,11 +1,109 @@
 'use strict';
 
 describe('fillTank', () => {
-  // const { fillTank } = require('./fillTank');
+  const { fillTank } = require('./fillTank');
 
-  it('should ', () => {
+  it('should be declared', () => {
+    expect(fillTank)
+      .toBeInstanceOf(Function);
+  });
+
+  it(`should order full tank if 'amount' is not given`, () => {
+    const customer = {
+      money: 3000,
+      vehicle: {
+        maxTankCapacity: 40,
+        fuelRemains: 8,
+      },
+    };
+    const expectedResult = 40;
+
+    fillTank(customer, 4);
+
+    expect(customer.vehicle.fuelRemains).toBe(expectedResult);
+  });
+
+  it(`should order full tank if 'amount'
+    is greater than the tank can accommodate`, () => {
+    const customer = {
+      money: 3000,
+      vehicle: {
+        maxTankCapacity: 40,
+        fuelRemains: 8,
+      },
+    };
+    const expectedResult = 40;
+
+    fillTank(customer, 4, 90);
+
+    expect(customer.vehicle.fuelRemains).toBe(expectedResult);
+  });
+
+  it(`should fill in only what the client can pay`, () => {
+    const customer = {
+      money: 3000,
+      vehicle: {
+        maxTankCapacity: 40,
+        fuelRemains: 8,
+      },
+    };
+    const expectedResult = {
+      money: 0,
+      vehicle: {
+        maxTankCapacity: 40,
+        fuelRemains: 13,
+      },
+    };
+
+    fillTank(customer, 600, 30);
```

Comment:
The expected result indicates that the customer's money should be reduced to 0, but the 'amount' parameter is 600, which might not deplete the customer's money entirely depending on the fuel price. Verify the logic and adjust the expected result or input parameters accordingly.